### DISCUSSION TITLE
PSA: Support for Windows 10 Start menu

### DISCUSSION THREAD

**Discussion by valinet** [valinet/ExplorerPatcher#920] [Announcements]
Hi

I have recently grown frustrated with the lack of flexibility regarding the Windows 11 Start menu. Unfortunately, officially, the old, XAML-based live tile Windows 10 Start menu is unavailable since build 22000.65 or so, although attempts have been made to resurrect it, the one I know about and follow being https://github.com/bbmaster123/10SM (❤️ many thanks to @bbmaster123 for the pointers showcased in his repository).

Taking that into consideration, I have decided to look into and attempt to provide a functional implementation for ExplorerPatcher. Mainly, I have tackled these areas:

* A method for having `StartMenuExperienceHost.exe` launch the old menu instead of the new one. In the builds where the legacy menu was available, there was a registry setting at `HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\Advanced\Start_ShowClassicMode` that the launcher looked into to determine which menu to launch. More recent versions skip that and just call `GetProductInfo`, a Win32 call which tells the type of Windows installation. For the XAML menu to launch, it expects to run on "Windows PPI Edition", what is commercially known as the Windows edition that ships with the Surface Hub. Previous workarounds for enabling the Windows Start menu meant using an older version of `StartMenuExperienceHost.exe` which looked for `Start_ShowClassicMode`. In ExplorerPatcher, a mechanism was implemented to trick `StartMenuExperienceHost.exe` into thinking it runs on the Surface Hub, thus enabling the menu.
* Fixing the context menus in the Start menu. These were broken. Resource files (`.pri`) from build 22000.51 are used instead, which work just fine, as well as an older version of `StartTileData.dll`. For this, file accesses are redirected to the different resources and libraries using IAT hooks in the Start menu process.
* Fixing a bug that introduced a 1-2 second delay when the menu was dismissed.

For now, the support for this feature is experimental. Quality updates will be made, as well as bug fixes, in the coming builds. The menu is functional, a solid alternative for what Windows 11 ships with.

**Please make sure you are connected to the Internet when installing this version in order to allow for the external resources needed for correct menu operation to download. If you don't have Internet access, the program will install, but you will need to rerun the setup file at a later time, when you have Internet access, so that it can download the necessary files.**

P.S. A blog post will be uploaded to https://www.valinet.ro describing the implementation in more technical details in the coming days.

Enjoy!

![image](https://user-images.githubusercontent.com/6503598/155619655-eecc6ca5-e4c2-4744-9800-98d1ddffb16d.png)

**Comment by Anixx** [comment]
Oh, that's good but also would not it be a bit easier to implement the Classic Start menu (like in Windows 95)? I think it should be quite easy. The OpenShell has this functionality, but 

- It does not support any arbitrary system theme. It has pre-defined skins for Classic theme, XP Luna, Vista, etc, but any new theme requires a newly designed skin. If one would implement it using standard system calls respecting the theming (like Quick Launch pop-up menus), then it would be compatible with any custom theme or any default Windows theme of any version (such as Windows 11 theme).

- It consumes a lot of memory and sometimes lags for many seconds to load.

So, would it be difficult you implement the basic functionality for the simplest start menu (without search, skin customization, etc), that would support arbitrary custom theming?

**Reply by valinet** [reply]
Man, difficult… it takes time. A lot of time. I mean, that should be pretty obvious if you follow this program from the beginning (which you do). Look on the commit history how development on Simple Window Switcher went, for example. Initially, I honestly thought it was a few days job, while ironing out most of the quirks took much more than that.

A Start menu in reality is even more complex than that. Parsing directory trees, gathering program icons, drawing correctly (supporting your use case where a lot of APIs don’t work), determining how to present itself as the Start menu, these all take time. Sure, it won’t take long to have something, but having something at production level quality will. And after you think you are done, people will demand more and more features and you’ll end up with something similar to what is already available.

And like, what’s the incentive where very solid alternatives are already available, OpenShell being one of them…? It is already a proven implementation that works great. 

> It does not support any arbitrary system theme. It has pre-defined skins for Classic theme, XP Luna, Vista, etc, but any new theme requires a newly designed skin.

This was probably so that it supports the various Start menu scenarios, like Windows 95 vs Windows 7 menu. Just speculating.

> It consumes a lot of memory and sometimes lags for many seconds to load.

Man, idk, you keep coming up with this, I don’t know what your expectations for memory are. Anyway, don’t necessarily go by what Task Manager reports, things there can be misleading. You also have a page file on the computer. And a lot of services that might be unused by most users eat much more than OpenShell. What unnecessary allocations have you seen specifically in OpenShell’s code to lead to this claims?

> but also would not it be a bit easier to implement the Classic Start menu (like in Windows 95)?

Easier as in easier to get going and roughly knowing what the journey reserves and knowing is doable, but harder as in effort and time taken.

> I think it should be quite easy. 

It’s the nth time I am encouraging you, open Visual Studio or whatever editor you want and get going. I don’t understand why you don’t do that already. In my opinion, you already waste (excuse the language) so much workarounding all kinds of workarounds via all the weird and less weird solutions out there for this classic theme, you spend so much time on it, that I don’t understand why don’t you tackle the problem from the root and start developing your own implementation, based ofc on what’s available in EP and whatever is out there. I say “waste” because I kind of consider all the complicated workarounds and hacks as a waste of time when it takes so long to make them work properly and when a dedicated solution could address the problem much better and be way cleaner, even if it involves actual programming. Computers are begging to be programmed, that’s their reason d’etre, so I don’t see the reluctance there. This menu is easy, then just start implementing it from scratch already. Good luck, there are plenty of resources available out there, so it should be a breeze!

**Reply by Anixx** [reply]
> supporting your use case where a lot of APIs don’t work

 My point is to use the generic Windows APIs that work under all themes.

**Reply by valinet** [reply]
> > supporting your use case where a lot of APIs don’t work
> 
> My point is to use the generic Windows APIs that work under all themes.

DrawThemeTextEx from uxtheme.dll is as generic as it gets, but it doesn’t work when themes are disabled (obviously), so it doesn’t work with the classic theme. There is no alternative to it (for some of its use cases, for example, drawing with DTT_COMPOSITED) in the Windows API that works when themes are disabled. In ExplorerPatcher, I had to manually implement this functionality (around +100 lines) to workaround DrawThemeTextEx not being available under classic theme. So yeah, not all generic APIs are available which is why many times a lot of programs have trouble under classic theme with the current method that enabled it: developers have to get out of their way and code specifically for it to have it supported.

**Reply by Anixx** [reply]
What API does this use?
![изображение](https://user-images.githubusercontent.com/1271736/155836423-6a508713-4df6-461c-afce-0da00b67c50b.png)
I think, it works under any theme.

**Reply by valinet** [reply]
What API for what? Many Win32 calls are made, each serving a purpose. Decompile the component responsible for that and check there.

What I mentioned above holds true. Certain things require certain workarounds and alternative implementations in order to work correctly under classic theme. DrawThemeTextEx is used to draw text with regards to the alpha channel for those times when you request the compositor to use a brush over the window before displaying it, for example. And that can theoretically work with classic theme as well, since in this hacky method that is used to enable it, the compositor still runs, but as I said, to fully accomplish the effect, one has to write their own DrawThemeTextEx since the built-in API doesn’t work anymore when themes are disabled.

**Reply by OrthodoxWindows** [reply]
I am not in expert on the subject, but I think the DrawThemeTextEx API is only little used, because it is rare that I meet programs partially incompatible with the classic theme, and is very rare that I meet programs completely incompatible with the classic theme.

**Reply by Anixx** [reply]
How the menus were implemented before the themes were introduced? Those menus still work with themes enabled, and look themed.

**Reply by valinet** [reply]
We are wasting time with this talk. No one here said you can’t implement things without themes enabled. It’s just that some things require additional work to support that, which is true, I gave the example of DrawThemeTextEx above. Rarely used? That’s one of the “mitigations” from EP for classic theme. That’s why the taskbar doesn’t show labels regularly: those are drawn using DrawThemeTextEx which doesn’t work when the classic theme is enabled. Same reasons button spacing is broken, other uxtheme APIs which do not work.

Gang-like upvoting serves no purpose. I replied quite a lot of stuff above, no one bothered to reply to that. I told you, I tell you again, give a shot at implementing it and you will see for yourself how hard/easy this is to do. I am telling you, from experience, you will have to do double work to properly support this classic theme method, and also acrylic/Mica/whatever backdrop brushes. The code for EP is public, I don’t speak from what I imagine, I don’t understand why we have to keep debating whether this is true or not.

Or what’s the question here really? Implement a Start menu from scratch now? Not in my plans. Neither do I have the time, nor the interest. I want to learn more interesting stuff or do something actually more useful than yet another Start menu clone when there are already plenty that work very well, one very solid one is open source as well, only to consume 5MB less RAM (apparently) and to support this classic theme in a **certain** way. Sorry.

P.S. Same reason I don’t want to integrate everything in 7+TT in EP conceptually. What’s the point in that? 7+TT is already a working, solid implementation that works great, what would such a port do?…

**Reply by Anixx** [reply]
So, how the things were implemented in Windows XP? One did net need any mitigations, and all programs written for XP work well both with modern themes and without the themes.

**Reply by valinet** [reply]
Better to compare to DOS... /s

Study a bit what a compositor does, for example, and how it works at a fundamental level, and you'll see that, despite both OSes displaying "windows", the way they do it and the capabilities and restrictions they have at their disposal are vastly different. I think I explained how it works in some post over here as well. Want to learn, fine, I am all in, want to continue a senseless argument, I am not willing to participate anymore. I keep asking you to look on things and learn about them, scratch things yourself etc, you keep ignoring that, never answer my questions, and then keep asking rhetoric questions. I don't have much free time each day and I don't want to waste it in an argument over something rather stupid. I don't understand what you want me to say abut it anymore. As I said, I always answered your questions, yet you most of the times ignore mine's.

**Comment by bbmaster123** [comment]
Very nice work! I'm glad I was able to help kick this off, little frustrated by my current lack of time tbh, but I am very much looking forward to reading through the blog post and all the little details, hopefully sometime next week <3

Keep it up man, you've been on absolute fire lately with all these other recently added features too !

PS: I hope you're doing well outside of this!

**Reply by valinet** [reply]
Thank you very much for the kind words, I appreciate it. Yeah, I hope to have a blog post ready in the coming days, hopefully.

I guess I am doing okay, could be better ofc, could also be much worse, so I guess I shouldn’t complain. Life in the past years was very favoring in regards to doing tons of work in the detriment of **everything** else, so yeah, there’s a general sense of tiredness that I have. 

I hope that you are doing fine as well on all plans and wish you all the best!

**Comment by Anixx** [comment]
I would like to test this feature (yes, it should work under Classic theme as in Win8.1 the start screen works), but the start button currently simply does not.

**Reply by valinet** [reply]
Does the Windows 10 Start menu work under classic theme in Windows 10? That is a better clue as to whether this has any chance of working out of the box than comparing to the ancient Windows 8.1 which had a totally different implementation.

**Reply by OrthodoxWindows** [reply]
In Windows 10, XAML start menu work under classic theme.

**Reply by OrthodoxWindows** [reply]
Just use the high contrast so that the menu respects roughly the colors of the classic theme.

**Comment by lixkote** [comment]
Can you also try to port the 21996 notification center?

**Reply by valinet** [reply]
Maybe, idk, is it still in there? Probably. I know in Insider Preview builds it still is in there, enabled for the PPI edition of Windows, and can be enabled by a setting in the registry, as far as I remember. Maybe a scheme similar to what is used for enabling the network flyouts could be used...? Idk, it has to be investigated, maybe I will look into it one day.

**Reply by SamuraisEpic** [reply]
Would this registry setting be "UseLiteLayout" in "HKLM\Software\Microsoft\Windows\Control Center", with DWORD 1?

**Reply by SamuraisEpic** [reply]
I'm working on a project that takes advantage of 10SM and ExplorerPatcher to return a more useful yet equally beautiful UI in Windows that also works for gaming.

**Reply by valinet** [reply]
As I said, I found that before, although idk if it was introduced in the 22000-based builds as well. I noticed that when I was looking in `twinui.pcshell.dll` from Insider Preview a while ago. I think I mentioned it in one of my blog posts as well.

**Reply by valinet** [reply]
Btw, `UseLiteLayout` is not only for the notification center. It also disables other things, like the `Win`+`A` action center. Keeping track of all of these is a mess.

**Reply by SamuraisEpic** [reply]
Oh ok, Thanks for the quick reply.

**Comment by OrthodoxWindows** [comment]
Very thank for this improvement ; I do not use this menu myself but I know people who uses it and appreciate it a lot. 
And I find myself the tiles very well realize (the icons respect the standard size of 32x32, which became rare).

**Reply by valinet** [reply]
You’re welcome.

**Comment by superjose** [comment]
I just came here to say thank you! I haven't upgraded to Win 11 because I use jump lists a lot, and the New start has a crippling experience. 

I've been following this and 10S project for a while, waiting it to be a little bit more stable before I upgrade. Seriously, thank you very much. I'm not short of words for all the effort placed in this project.

**Reply by valinet** [reply]
Thank you, and I will direct the words to @bbmaster123 as well who did the original research.

Yeah, I do believe in no bullshit open development. It may not be much, but it’s there, it’s open, it’s freely available and published today. I may not believe in much, but what I do, I totally stand by it.

Thanks again

**Reply by superjose** [reply]
Awesome!! 🤗🤗🤗

**Comment by novatiberium** [comment]
One huge step for this project! Hopefully we can add support for other languages soon.

**Reply by valinet** [reply]
Other languages where?

**Reply by novatiberium** [reply]
For the start menu. Currently half of it is shown in English if you are running Windows 11 with EP in a foreign language (like some parts, but shutdown menu is translated)
![תמונה](https://user-images.githubusercontent.com/68118732/155912349-e5567782-cea2-4d6a-b297-e8f1b5e155aa.png)

**Reply by valinet** [reply]
Go to “C:\Windows\ SystemApps\Microsoft.Windows.StartMenuExperienceHost_cw5n1h2txyewy” and make a folder in there with your locale name (there is already a folder called “en-US”, you need to create a similar one: language-region, for e.g. “ro-RO”). Grab “StartTileData.dll.mui from C:\Windows\System32\ro-RO, rename it to StartTileDataLegacy.dll.mui and place it in that folder and the elements should now be translated (ofc, restart File Explorer or the computer for it to have effect).

**Reply by novatiberium** [reply]
![תמונה](https://user-images.githubusercontent.com/68118732/156235912-b9ea3c6a-c504-41e4-a96f-a9572b707360.png)
![תמונה](https://user-images.githubusercontent.com/68118732/156235925-f99abb5a-9c74-4861-9cbd-b3fa5a6a32c2.png)
Despite the fact that I followed instructions, elements remain untranslated.

**Reply by valinet** [reply]
Yeah, idk, maybe the mui file from 22000.51 should be used. Idk. It’s not a big deal anyway, at least the thing works.

**Reply by valinet** [reply]
Okay, yeah, I remembered. You also need to place in the `C:\Windows\ SystemApps\Microsoft.Windows.StartMenuExperienceHost_cw5n1h2txyewy\pri2` folder a copy of `C:\Windows\SystemResources\Windows.UI.ShellCommon\pris\Windows.UI.ShellCommon.ro-RO.pri` for example from build 22000.51. That I think contains the rest of the translations. Mandatory from build 22000.51, as the newer ones do not work and just crash the menu.

**Reply by Lyhtande** [reply]
Hello, this worked for me with the de-DE files from from build 22000.51.
But since the last Explorer Patcher update (22621.1992.56.1) everything switched back to English. Could someone please provide me with the German files for 22621.1992.56.1? Maybe the de-DE files for `C:\Windows\ SystemApps\Microsoft.Windows.StartMenuExperienceHost_cw5n1h2txyewy\pris2\Windows.UI.ShellCommon.de-DE.pri` and  `C:\Windows\ SystemApps\Microsoft.Windows.StartMenuExperienceHost_cw5n1h2txyewy\de-DE\StartTileDataLegacy.dll.mui` are not enough anymore to get a German languaged Start Menu?

**Comment by Thornskade** [comment]
I was excited to try it but it doesn't work on my end. The start menu will not open, or sometimes after enabling it it will show up once and then never again, so it's there and does work but just refuses to open up

I have no idea why that happens

EDIT: Yeah, okay, so this is very reproducable behavior. When I open the Properties windows and select the 11 menu, it works fine every time. When I select the 10 menu and click on Start or press the Windows button, the 10 menu will take some unusually long time to show up, and only appear that one time. When I close it and try to reopen it it won't work until I switch back to 11 and then back to 10

**Reply by hiranokite** [reply]
if you upgraded from 10 to 11 with "Use Start full screen" enabled, follow the steps to disable this using regedit, mentioned here #933.
for now full screen is not working.

**Reply by Thornskade** [reply]
Oh, that's why. Thank you very much for informing me. I'll stick to Start11 for the time being, but I do prefer the look and animations of the OG start menu

**Reply by valinet** [reply]
> if you upgraded from 10 to 11 with "Use Start full screen" enabled, follow the steps to disable this using regedit, mentioned here https://github.com/valinet/ExplorerPatcher/discussions/933.
for now full screen is not working.

I fixed the Start menu to work in full screen in latest pre-release (42.23).

**Reply by Thornskade** [reply]
> I fixed the Start menu to work in full screen in latest pre-release (42.23).

I just tried it. That's awesome it works now, however it has some issues with its dimensions. These seem to be taken from you main monitor and so if your taskbar is on a secondary display, the dimensions may be entirely wrong (it's unusable on my 4:3 monitor and my vertical 16:9 monitor, for example)

EDIT: I just found out there's a workaround. When you enable "open on display with cursor" then the dimensions will be correct, but of course that means when you play a game the start menu will overlay on top of it, and potentially fail to do so if your game is in fullscreen mode; for my use case that's not exactly what I want, but it's an improvement for sure

**Comment by SamuraisEpic** [comment]
Does start11 do fullscreen? I'm pretty sure it doesn't. The normal non fullscreen menu work 100% fine tho


________________________________

**Reply by valinet** [reply]
Implemented support for full screen Start in latest pre-releases.

**Reply by SamuraisEpic** [reply]
Ok, thank you.

**Comment by sdw4938** [comment]
Live tiles aren't working for me

**Reply by SamuraisEpic** [reply]
Live tiles should work, though you need LiveTilesAnywhere from the store and configure it to use them

**Comment by Karell2013** [comment]
Hello, I would like to know when the old Windows 10 Action or Notification Center is introduced?

**Comment by TheNFSModder217132** [comment]
does not work for other languages in some places

**Reply by valinet** [reply]
There is already a post and a solution for this in some post up here.

**Comment by sdw4938** [comment]
Does the rounded win10 start menu work on 19041?

**Reply by SamuraisEpic** [reply]
I doubt it, but you can see. and why are you staying on 19041? Is it the clear tile backgrounds? Because those can be disabled (at least I think so)

**Reply by valinet** [reply]
Technically it would work, practically not at the moment because support on Windows 10 is more cumbersome... at some point it will get there as well.

**Comment by ghost** [comment]
signed up in here just to say i really admire your work on this and also i would like to ask a question or may be a request regarding my clean installed recent windows 11 i do not have the option in windows 10 menu to disable "Show App List in Start menu" so i can have only the pinned items on the start. All the work you have done here makes me enjoy using windows 11 more than i thought i would. Im using the latest windows 11 and most recent pre-released Explorer Patcher

**Reply by sdw4938** [reply]
> 1. Click the Start button, type regedit, and then press Enter.
> 2. If prompted, choose Yes or enter the administrator password.
> 3. On the left-side pane, navigate to the path below.
> 4. HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\Explorer
> 5. Right-click Explorer and then select New > DWORD (32-bit) Value.
> 6. Name the value as NoStartMenuMorePrograms and then hit Enter.
> 7. Double-click NoStartMenuMorePrograms, set the Value data to 3 and then click OK.
> 8. Close the Registry Editor and then restart explorer.

Sources: https://answers.microsoft.com/en-us/windows/forum/all/remove-all-start-menu-icons-in-windows-10/7c9cf9ed-d034-4a3a-81e1-879fc9de41c2
https://appuals.com/add-or-remove-all-apps-list-in-start-menu-on-windows-10/
![Screenshot 2022-03-02 213817](https://user-images.githubusercontent.com/76626288/156485750-35c2ffe8-7187-482c-874a-02ff85be11d6.png)
![Screenshot 2022-03-02 213832](https://user-images.githubusercontent.com/76626288/156485751-3dc01238-04ec-4ead-95ae-f8e46ec807bd.png)

**Reply by valinet** [reply]
Added an option for this in the latest pre-release. Thanks for mentioning it. The Windows 10 menu was indeed proper, catering to most of these legacy GPOs, as opposed to the new menu which ignores pretty much anything, in a one-size-fits-all approach.

**Reply by ghost** [reply]
Huge thanks to you guys for making this possible. And we all expected a lot from windows at first and they made some good decisions but the start menu was the only downgrade. Not Anymore😊 also i have been using this patch for a week and havent run into any bugs. I will folowing this project very closely. I Admire the work thats been done here. Setting up my desktop and this is how it looks so far. 
![Screenshot 2022-03-03 121032](https://user-images.githubusercontent.com/100816109/156508990-146f430d-0230-4098-aef5-0c83e12eba5d.png)

**Comment by hiranokite** [comment]
would there be any way to do the search inside the windows 10 start menu, without having to open the 11 search menu?

**Reply by valinet** [reply]
No, afaik, there’s not.

**Comment by Anixx** [comment]
Is it possible to use Win10 start menu with OpenShell button (as the default button simply does not work on Classic theme)?

**Reply by OrthodoxWindows** [reply]
Maybe the problem could be solved by going to the openshell settings, then in the "Control" tab, then "Left click opens, then "Windows start menu".
![seamonkey_gQzccvemwK](https://user-images.githubusercontent.com/100134023/156819128-ebaaf94c-ea84-4ab4-9d6b-187b03913c1b.png)

**Comment by Anixx** [comment]
It looks now that under any theme neither the Win10 start menu, not Win11 start menu appears for me... Have no idea what can be the cause...

**Comment by qbiecom** [comment]
Fabulous work! I've uninstalled Start11 and am using this.

Is there any way the quick links to settings, documents etc.. can be added in as they are on the Win 10 start menu?

![image](https://user-images.githubusercontent.com/6334619/156743857-e6223b75-29c8-4c6c-9600-8bff1d850212.png)

**Reply by Poopooracoocoo** [reply]
I added them just from the Settings app

**Reply by qbiecom** [reply]
Aha, got it! Thanks :)

**Comment by Abolohit** [comment]
Thanks for your continued development of ExplorerPatcher, and taking off from where BBMaster left off bringing Windows 10 start menu to 11. It works very nicely in 11. 

I have found a small issue though. Most of the live tiles work fine, like that of Weather App, Mail and Calender, etc. 

But, the Live tile of Groove Music gets stuck after updating the first time. Could it be an issue with Groove music itself? In the windows 11 action centre, the groove music player flyout above the volume and network settings flyout updates the currently playing track correctly. 

However, to get the start menu live tile to update, I have to turn off the live tile and turn it on again, then it updates once before getting stuck.

I am using the latest pre release build of Explorer Patcher, and the latest version of Groove Music just before it is replaced by Media Player

**Comment by sxxov** [comment]
Love this project, couldn't imagine Windows 11 without it; this update bolsters that to the moon. However, I do have a few gripes, mainly how the Windows 10 start menu coming in is only animated with its contents, as opposed to the additional fade slide-in in Windows 10. The other thing is how live tiles used to update with a wipe-in of the new content from the bottom of the tile, but instead now only the old content wipes-out & the new content flashes in afterwards.

Again, quite minor gripes, can't understate my appreciation for this project!

**Reply by valinet** [reply]
Yeah, it has some minor glitches, Microsoft broke some things, but at least I got it to being usable. Fading in/out is actually broken for some reason, probably they changed the backend a bit, it's the reason most probably there was that famous delay when dismissing the menu that I had to take care of as well. It's pretty opaque, hard to figure out how it really works, yet I am glad it does as is even.

**Reply by sxxov** [reply]
I see, thank you for the explanation, hope your magic will be able to bless it the right way eventually!

Also, just a note that there is still a delay if the start menu is dismissed via the `Esc` key, otherwise it's super responsive.

**Reply by valinet** [reply]
> I see, thank you for the explanation

You're welcome

> Also, just a note that there is still a delay if the start menu is dismissed via the Esc key, otherwise it's super responsive.

Hmm, I don't experience any delay with `Esc` key. Or perhaps you don't know how bad the delay was out of the box (1-2 seconds). If it's a touch slower than pressing the `Win` key, that's nothing, really.

> hope your magic will be able to bless it the right way eventually!

It's no magic, it's just an obscene amount of time spent looking on stuff and trying to figure it out. I wish it were magic, as in saying some spell and then you get it for free. Or maybe, idk, I guess it wouldn't be as rewarding if it were that easy to obtain.

**Reply by sxxov** [reply]
> Or perhaps you don't know how bad the delay was out of the box (1-2 seconds).

Oof that's rough haha; this is my first time experiencing the Windows 10 start menu on 11 so I definitely had no idea.

> I wish it were magic, as in saying some spell and then you get it for free. Or maybe, idk, I guess it wouldn't be as rewarding if it were that easy to obtain.

If Harry Potter has taught me anything, it's that magic comes with more stupid shit than one would expect without it (;;

**Comment by Aster-the-Med-Stu** [comment]
The Windows 10 start just refuses to appear (crashes) on my Surface Go. Event log has this in entry:

```
Faulting application name: StartMenuExperienceHost.exe, version: 0.0.0.0, time stamp: 0xae77557b
Faulting module name: Windows.UI.Xaml.dll, version: 10.0.22000.493, time stamp: 0x632525e9
Exception code: 0xc000027b
Fault offset: 0x0000000000834af0
Faulting process id: 0x2884
Faulting application start time: 0x01d8311fb4eb14b0
Faulting application path: C:\Windows\SystemApps\Microsoft.Windows.StartMenuExperienceHost_cw5n1h2txyewy\StartMenuExperienceHost.exe
Faulting module path: C:\Windows\System32\Windows.UI.Xaml.dll
Report Id: e68877d6-66d2-4e1a-b560-9a5dc7fe834c
Faulting package full name: Microsoft.Windows.StartMenuExperienceHost_10.0.22000.37_neutral_neutral_cw5n1h2txyewy
Faulting package-relative application ID: App
```

[Windows Error Reporting generated file is attched.](https://github.com/valinet/ExplorerPatcher/files/8192044/Report.wer.zip) Any hints for why this is happening?

**Reply by valinet** [reply]
Yeah, idk. Restart the computer. Other than that, the debugger is one's best friend. I cannot infer anything from that log. A stack trace would have been better to look at, though it doesn't guarantee anything.
